### DISCUSSION TITLE
Has the save as text file feature been removed from 3.5 onwards?

### DISCUSSION THREAD

**Discussion by jamiecropley** [sumatrapdfreader/sumatrapdf#3823] [General]
Has the save as text file feature been removed from 3.5 onwards?

**Version 3.4.6 - 64 Bit Windows 10:**

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/17186004/74599986-a2b8-4217-9b04-23da4706ad1e)

**Version 3.5.2 - 64 Bit Windows 10:**

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/17186004/daedd216-7b09-41b7-954b-32dcdb55456e)

Why was it removed and will the feature be coming back?

**Comment by kjk** [comment]
yes. it was very obscure. might come back as an explicit command but then I would like to understand what would be desired functionality for this.

Just save all extracted text to a .txt file?
Ability to extract from one (or more) pages?
Ability to copy to clipboard?

**Reply by jamiecropley** [reply]
So personally I use it for saving open source academic papers for interpretation into an LLM / AI model, its very useful for me personally like this. For now I have downloaded version 3.4.6 again and I can use later versions if I need to in future. Please reconsider if possible though. I would just use something like pandoc but it don't support it. 

Saving to the text file is best for my use case.

**Reply by miloush** [reply]
I was using this feature and would welcome an explicit command; happy to contribute.

Just save all extracted text to a .txt file is good enough. It is a bit unfortunate to have whole features removed because they don't work in a few cases as expected.

**Reply by jamiecropley** [reply]
> I was using this feature and would welcome an explicit command; happy to contribute.
> 
> Just save all extracted text to a .txt file is good enough. It is a bit unfortunate to have whole features removed because they don't work in a few cases as expected.

I no longer use windows, in fact I use a software called Skim now. But anyway if you use Apache Tika, it works much better for this purpose.

**Comment by jamiecropley** [comment]
Can you respond and let me know if you add this back in some day so then I can re-update to the latest version. Thanks.

**Reply by GitHubRulesOK** [reply]
@jamiecropley 
There can be issues with copy and paste as its not always what it could be see the current discussion in #3844 

What is generally reliable is poppler pdf to text which can extract by pages or by area selection of page/pages

SumatraPDF can send the current page number or filename of the pdf in view and that can then invoke pdftotext so as to most often get a good result.

The use of **-layout** ensures indentation, (PDF does NOT use tabs). The only downside is that like every other method line feeds have to be injected where there are NONE in a PDF

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/b47f4622-9868-4f0f-8940-2f73270fe86d)

I am using the last known 32bit version 2022-02 however if you are on 64bit you can use this years 23-11 binary version from here on github https://github.com/oschwartz10612/poppler-windows

ExternalViewers [
[
CommandLine = "C:\Users\lez\Downloads\Apps\PDF\poppler\22.02\pdftotext.exe" -nopgbrk -layout -enc UTF-8 -f %p -l %p %1 
name=Extract page &text
]
]

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/900587c4-4b9f-4c9a-9cb0-d0629277dc33)

if you want to set a different name for output (currently as shown the .txt will be alongside the .pdf) OR if you want to run other commands like open in Notepad etc. then its best to use a cmd file to very simply capture the %p and %1 input for manipulation by multiple other windows commands.

**Comment by jamiecropley** [comment]
Ok so I found a good solution earlier, as I am mainly using it for AI data sets and such, I found Apache Tika well suited to my needs.

**Reply by GitHubRulesOK** [reply]
interesting it uses MuTool so of course you could use that too as a replacement to poppler 
cmd Mutool convert pdf to stext or html etc etc.

**Comment by u07** [comment]
>  I would like to understand what would be desired functionality for this.

1) Feeding the text to other software like scripts, statistic analyzers. 
2) Manual text processing in Notepad++ with all it's editing power like similarities highlight. 
3) Bulk quotation - Sumatra lacks shift+click (set selection end), ctrl+select (multiselect) and all that cool stuff.
4) regexp search
5) text comparison 
6) grammar check in word

Can't imagine, what made you get it removed. Please, revert if possible.

**Reply by jamiecropley** [reply]
You ignored the above comment, just use apache tika. Additionally the entire thread is closed and resolved.
